workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/ModalContainerView.kt

```diff
@@ -0,0 +1,168 @@
+package com.squareup.workflow1.ui
+
+import android.app.Dialog
+import android.content.Context
+import android.os.Bundle
+import android.os.Parcel
+import android.os.Parcelable
+import android.os.Parcelable.Creator
+import android.util.AttributeSet
+import android.view.View
+import android.view.ViewGroup
+import android.view.ViewGroup.LayoutParams.MATCH_PARENT
+import android.widget.FrameLayout
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.Lifecycle.Event.ON_DESTROY
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+
+@WorkflowUiExperimentalApi
+class ModalContainerView
+@JvmOverloads constructor(
+  context: Context,
+  attributeSet: AttributeSet? = null,
+  defStyle: Int = 0,
+  defStyleRes: Int = 0
+) : FrameLayout(context, attributeSet, defStyle, defStyleRes) {
+
+  private val baseView: WorkflowViewStub = WorkflowViewStub(context).also {
+    addView(it, ViewGroup.LayoutParams(MATCH_PARENT, MATCH_PARENT))
+  }
+
+  private var dialogs: List<Dialog> = emptyList()
+
+  private fun update(
+    newScreen: ModalContainerViewRendering<*, *>,
+    viewEnvironment: ViewEnvironment
+  ) {
+    baseView.show(newScreen.beneathModals, viewEnvironment)
+
+    val updateDialogs = mutableListOf<Dialog>()
+    for ((i, modalRendering) in newScreen.modals.withIndex()) {
+      updateDialogs +=
+        if (i < dialogs.size && compatible(dialogs[i].getDisplaying()!!, modalRendering)) {
+          dialogs[i].apply { display(modalRendering, viewEnvironment) }
```

Comment:
AFAIK alert dialogs can't be updated in place. Perhaps AlertModalRendering should implement Compatible?